pkg/applicationdefinitions/application_catalog.go

```diff
@@ -150,8 +149,6 @@ func updateApplicationDefinition(appDef *appskubermaticv1.ApplicationDefinition,
 	for i := range appDef.Spec.Versions {
 		if sysAppConfig.HelmRepository != "" {
 			appDef.Spec.Versions[i].Template.Source.Helm.URL = registry.ToOCIURL(sysAppConfig.HelmRepository)
-		} else {
-			appDef.Spec.Versions[i].Template.Source.Helm.URL = defaulting.SystemApplicationsHelmRepositoryFromConfig(&sysAppConfig)
```

Comment:
Note for reviewer:

We are now using YAMLs to store these ApplicationDefinitions and use explicit URLs there. So there is no need for this defaulting through code.